app/controllers/api/v1x0/icons_controller.rb

```diff
@@ -23,30 +23,36 @@ def update
       end
 
       def raw_icon
-        image = find_icon(params.permit(:icon_id, :portfolio_item_id)).image.decoded_image
+        image = find_icon(params.permit(:icon_id, :portfolio_item_id, :portfolio_id)).image.decoded_image
         send_data(image,
                   :type        => MimeMagic.by_magic(image).type,
                   :disposition => 'inline')
       end
 
       def override_icon
-        overriden_icon = Catalog::OverrideIcon.new(params.require(:icon_id), params.require(:portfolio_item_id))
+        overriden_icon = Catalog::OverrideIcon.new(params.require(:icon_id), params.permit(:portfolio_id, :portfolio_item_id))
         render :json => overriden_icon.process.icon
       end
 
       private
 
       def icon_params
-        params.require([:content, :portfolio_item_id])
+        params.require(:content)
         icon_patch_params
       end
 
       def icon_patch_params
-        params.permit(:content, :source_ref, :source_id, :portfolio_item_id, :id)
+        params.permit(:content, :source_ref, :source_id, :portfolio_item_id, :portfolio_id)
       end
 
       def find_icon(ids)
-        ids[:icon_id].present? ? Icon.find(ids[:icon_id]) : Icon.find_by!(:portfolio_item_id => ids[:portfolio_item_id])
+        if ids[:icon_id].present?
+          Icon.find(ids[:icon_id])
+        elsif ids[:portfolio_item_id].present?
+          PortfolioItem.find(ids[:portfolio_item_id]).icons&.first
+        elsif ids[:portfolio_id].present?
+          Portfolio.find(ids[:portfolio_id]).icons&.first
+        end
```

Comment:
After looking into how I'd test that exception, there isn't a way it would get hit. Currently the only way to hit this method is by the `#raw_icon` operation, which can only be called from these endpoints:
- `/icons/{icon_id}/icon_data`
- `/portfolio_items/{portfolio_item_id}/icon`
- `/portfolios/{portfolio_id}/icon`

So one of these 3 IDs will always be present. I'm not sure of a cleaner way to say "one of these 3 strings will be there", suggestions?